[Detection Engine] Prevent test data from rolling outside of Cypress rules' query

rylnd opened the pull request:
## Summary

Our tests rely on static test data, and our cypress suite was running rules with a large lookback (50000h) against them. Recently, tests have begun failing because we've exceeded 50000h since the data's `@timestamp`. This PR updates the cypress rules to use a fixed `from` field wherever possible, in order to "fix" that query window's starting point, ensuring that this issue does not recur. This is already how our integration tests are creating similar rules, so as an added bonus we're now more consistent across test suites, as well.

This unskips and thus resolves the following issues:

* https://github.com/elastic/kibana/issues/201334
* https://github.com/elastic/kibana/issues/220822
* https://github.com/elastic/kibana/issues/207913
* https://github.com/elastic/kibana/issues/199905

Note that there may be more instances of skipped tests that were failing due to this issue (with e.g. data with an earlier date than the `2019-09-02` here); the above tests were just the most obvious and most recent (within the last week).

### Flaky Runner
* Detection Engine Cypress (x150): https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8291

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

<!--ONMERGE {"backportTargets":["8.17","8.18","8.19","9.0"]} ONMERGE-->

rylnd committed:
Unskip tests failing due to stale test data

See previous commit.

rylnd commented:
/ci

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/300782)
* Commit: 9e7b53e36799890f47677782b553adb0f684d9d5

### Failed CI Steps
* [FTR Configs #26](https://buildkite.com/elastic/kibana-pull-request/builds/300782#0196e9c4-7d80-4fdc-b5aa-83da1e0c42ff)
* [FTR Configs #28](https://buildkite.com/elastic/kibana-pull-request/builds/300782#0196e9c4-7d82-4ee0-8e47-ca65c6b16354)
* [FTR Configs #46](https://buildkite.com/elastic/kibana-pull-request/builds/300782#0196e9c4-7d8c-421a-b93c-e4bd29d6a996)
* [FTR Configs #122](https://buildkite.com/elastic/kibana-pull-request/builds/300782#0196e9c4-7dbd-441a-8f87-6840ff6a8972)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/300782#0196e9c4-7dbd-441a-8f87-6840ff6a8972) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/300782/jobs/0196e9c4-7dbd-441a-8f87-6840ff6a8972/artifacts/0196e9d9-0c48-4de9-8344-aaa92fd372ed) FTR Configs #<span></span>122 / Cloud Security Posture Findings Page onboarding Misconfigurations - clicking on the `No integrations installed` prompt action button - `install cloud posture integration`: navigates to the CSPM integration installation page
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/300782#0196e9c4-7d82-4ee0-8e47-ca65c6b16354) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/300782/jobs/0196e9c4-7d82-4ee0-8e47-ca65c6b16354/artifacts/0196e9ec-558c-4d2f-af5b-e33f538a5e2b) FTR Configs #<span></span>28 / console app misc console behavior keyboard shortcuts should execute the request when Ctrl+Enter is pressed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/300782#0196e9c4-7d80-4fdc-b5aa-83da1e0c42ff) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/300782/jobs/0196e9c4-7d80-4fdc-b5aa-83da1e0c42ff/artifacts/0196e9e6-90e6-49d1-92b5-57b22ebeddc7) FTR Configs #<span></span>26 / Embeddable Alerts Table Embeddable alerts panel "before all" hook for "should only show alerts from the observability area (o11y+stack) when selecting it"

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300782","number":300782,"commit":"9e7b53e36799890f47677782b553adb0f684d9d5"}],"number":300782}
buildkite-pr-comment-->

rylnd committed:
Update cypress DE rules to query from a fixed (starting) time frame

As opposed to a rolling window of 50,000h (~5.7 years), this updates our
cypress-created rules to query a window with a fixed starting point.

The rolling window was problematic because our static test data will
eventually fall outside of this window. While the fixed `from` means
that we will technically be querying an increasing window over time,
this should have little practical effect as the data should remain
equally sparse.

This is also how our integration tests generate rules, so cypress rules
are now aligned in that approach, as well.

rylnd committed:
Remove one more instance of a rolling-windowed test

This should not be needed and only lends us to future failures.

rylnd committed:
Unskip cypress tests failing due to rule not generating alerts

These have been fixed by fixing the window in which the rules query, so
that our static data will no longer be able to roll "outside" of that
window (which was causing these failures).

There may be other instances of this type of failure, but these are the
obvious ones from the last few days.

rylnd committed:
Merge branch 'main' into rylnd/fix_outdated_test_data

 Conflicts:
	x-pack/test/security_solution_cypress/cypress/e2e/detection_response/detection_engine/exceptions/alerts_table_flow/rule_exceptions/closing_all_matching_alerts.cy.ts
	x-pack/test/security_solution_cypress/cypress/e2e/detection_response/detection_engine/exceptions/rule_details_flow/add_edit_exception.cy.ts

rylnd commented:
/ci

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/301522)
* Commit: 80901e8a33011b0818dd33604c8f617a12b19e4a

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/301522#0196f485-d0ab-424c-bb6a-95a8c9eef9a0)

### History
* :yellow_heart: [Build #300782](https://buildkite.com/elastic/kibana-pull-request/builds/300782) was flaky 9e7b53e36799890f47677782b553adb0f684d9d5

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301522","number":301522,"commit":"80901e8a33011b0818dd33604c8f617a12b19e4a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300782","number":300782,"commit":"9e7b53e36799890f47677782b553adb0f684d9d5"}],"number":301522}
buildkite-pr-comment-->

rylnd committed:
Fix parameter typo

rylnd commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/301532)
* Commit: 4d93afa53ff886347e6d3d46bd06c0a1ce40491d

### Failed CI Steps
* [Serverless Rule Management - Security Solution Cypress Tests #5](https://buildkite.com/elastic/kibana-pull-request/builds/301532#0196f4b5-10d3-4cbb-985c-3c3e5b1c2e87)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/301532#0196f4b5-10d3-4cbb-985c-3c3e5b1c2e87) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/301532/jobs/0196f4b5-10d3-4cbb-985c-3c3e5b1c2e87/artifacts/0196f4e2-b16f-4183-977e-188bebd8f449) Serverless Rule Management - Security Solution Cypress Tests #<span></span>5 / Bulk editing index patterns of rules with a data view only "before each" hook for "Add index patterns to custom rules with configured data view: all rules are skipped" "before each" hook for "Add index patterns to custom rules with configured data view: all rules are skipped"

### History
* :broken_heart: [Build #301522](https://buildkite.com/elastic/kibana-pull-request/builds/301522) failed 80901e8a33011b0818dd33604c8f617a12b19e4a
* :yellow_heart: [Build #300782](https://buildkite.com/elastic/kibana-pull-request/builds/300782) was flaky 9e7b53e36799890f47677782b553adb0f684d9d5

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301532","number":301532,"commit":"4d93afa53ff886347e6d3d46bd06c0a1ce40491d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301522","number":301522,"commit":"80901e8a33011b0818dd33604c8f617a12b19e4a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300782","number":300782,"commit":"9e7b53e36799890f47677782b553adb0f684d9d5"}],"number":301532}
buildkite-pr-comment-->

rylnd committed:
Set new_terms history window to 150 years

The new_terms validation requires that the history window be greater
than the `from`, which is currently set to 1900.

If you're a developer reading this commit in 2050, I apologize.

rylnd commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/301546)
* Commit: 2563ef75954d15af2baf5a20941c0ad6c40c5627

### Failed CI Steps
* [Detection Engine - Security Solution Cypress Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/301546#0196f502-21d6-4ae5-840b-97738f246f6e)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/301546#0196f502-21d6-4ae5-840b-97738f246f6e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/301546/jobs/0196f502-21d6-4ae5-840b-97738f246f6e/artifacts/0196f53d-7a31-4fe6-925d-da33728e053b) Detection Engine - Security Solution Cypress Tests #<span></span>3 / Indicator Match - Rule Creation Detection rules, Indicator Match Generating signals Creates and enables a new Indicator Match rule Creates and enables a new Indicator Match rule

### History
* :broken_heart: [Build #301532](https://buildkite.com/elastic/kibana-pull-request/builds/301532) failed 4d93afa53ff886347e6d3d46bd06c0a1ce40491d
* :broken_heart: [Build #301522](https://buildkite.com/elastic/kibana-pull-request/builds/301522) failed 80901e8a33011b0818dd33604c8f617a12b19e4a
* :yellow_heart: [Build #300782](https://buildkite.com/elastic/kibana-pull-request/builds/300782) was flaky 9e7b53e36799890f47677782b553adb0f684d9d5

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301546","number":301546,"commit":"2563ef75954d15af2baf5a20941c0ad6c40c5627"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301532","number":301532,"commit":"4d93afa53ff886347e6d3d46bd06c0a1ce40491d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301522","number":301522,"commit":"80901e8a33011b0818dd33604c8f617a12b19e4a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300782","number":300782,"commit":"9e7b53e36799890f47677782b553adb0f684d9d5"}],"number":301546}
buildkite-pr-comment-->

rylnd committed:
Remove tricky assertion from Indicator Match test

In moving the default `from` for rules to a _static, absolute_ value (to
address a separate class of failures), this assertion started failing.
In investigating the cause, I found that we both provide and assert on
values based on the duration _from the current time_. For relative
values, i.e. `now-5m`, this would always equate to 5 minutes, but with
an absolute value it was now increasing as time passed, and subsequent
calls with the same absolute value would likely never return the same
value.

I debated trying to account for this with some "fuzzy" matching (i.e.
"interval must be between the provided value and (provided value + some
duration)"), but that would still depend on the time between calls not
exceeding that value.

Ultimately, since:

* this assertion is ultimately just asserting that the value we provide
to the form is the value that is saved, and
* we test this field for other rule types, with more reliable values, and
* there is no indicator-match specific logic to this field,

I decided that keeping this particular assertion was not worth the
effort.

rylnd commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/301563)
* Commit: ce45e5281a5b6e6234275e748cae62d45a5b75cf

### Failed CI Steps
* [Detection Engine - Security Solution Cypress Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/301563#0196f614-526a-432f-afa3-78e879200ad5)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/301563#0196f614-526a-432f-afa3-78e879200ad5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/301563/jobs/0196f614-526a-432f-afa3-78e879200ad5/artifacts/0196f644-8be9-441f-b7f7-2c7c988c3a0e) Detection Engine - Security Solution Cypress Tests #<span></span>3 / Indicator Match - Rule Creation Detection rules, Indicator Match Generating signals Creates and enables a new Indicator Match rule Creates and enables a new Indicator Match rule

### History
* :broken_heart: [Build #301546](https://buildkite.com/elastic/kibana-pull-request/builds/301546) failed 2563ef75954d15af2baf5a20941c0ad6c40c5627
* :broken_heart: [Build #301532](https://buildkite.com/elastic/kibana-pull-request/builds/301532) failed 4d93afa53ff886347e6d3d46bd06c0a1ce40491d
* :broken_heart: [Build #301522](https://buildkite.com/elastic/kibana-pull-request/builds/301522) failed 80901e8a33011b0818dd33604c8f617a12b19e4a
* :yellow_heart: [Build #300782](https://buildkite.com/elastic/kibana-pull-request/builds/300782) was flaky 9e7b53e36799890f47677782b553adb0f684d9d5

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301563","number":301563,"commit":"ce45e5281a5b6e6234275e748cae62d45a5b75cf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301546","number":301546,"commit":"2563ef75954d15af2baf5a20941c0ad6c40c5627"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301532","number":301532,"commit":"4d93afa53ff886347e6d3d46bd06c0a1ce40491d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301522","number":301522,"commit":"80901e8a33011b0818dd33604c8f617a12b19e4a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300782","number":300782,"commit":"9e7b53e36799890f47677782b553adb0f684d9d5"}],"number":301563}
buildkite-pr-comment-->

rylnd committed:
Use a more recent `from` for indicator match rules

Because this duration is imbedded as a filter in the generated ES
queries, 125 years in seconds was causing `number_format_exceptions`

rylnd commented:
/ci

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/301759)
* Commit: 037a7b2727c0d2f106ab84783a847eeff2005f5d

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #301563](https://buildkite.com/elastic/kibana-pull-request/builds/301563) failed ce45e5281a5b6e6234275e748cae62d45a5b75cf
* :broken_heart: [Build #301546](https://buildkite.com/elastic/kibana-pull-request/builds/301546) failed 2563ef75954d15af2baf5a20941c0ad6c40c5627
* :broken_heart: [Build #301532](https://buildkite.com/elastic/kibana-pull-request/builds/301532) failed 4d93afa53ff886347e6d3d46bd06c0a1ce40491d
* :broken_heart: [Build #301522](https://buildkite.com/elastic/kibana-pull-request/builds/301522) failed 80901e8a33011b0818dd33604c8f617a12b19e4a
* :yellow_heart: [Build #300782](https://buildkite.com/elastic/kibana-pull-request/builds/300782) was flaky 9e7b53e36799890f47677782b553adb0f684d9d5

cc @rylnd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301759","number":301759,"commit":"037a7b2727c0d2f106ab84783a847eeff2005f5d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301563","number":301563,"commit":"ce45e5281a5b6e6234275e748cae62d45a5b75cf"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301546","number":301546,"commit":"2563ef75954d15af2baf5a20941c0ad6c40c5627"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301532","number":301532,"commit":"4d93afa53ff886347e6d3d46bd06c0a1ce40491d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/301522","number":301522,"commit":"80901e8a33011b0818dd33604c8f617a12b19e4a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/300782","number":300782,"commit":"9e7b53e36799890f47677782b553adb0f684d9d5"}],"number":301759}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/security-detection-engine (Team:Detection Engine)

kibanamachine commented:
## Flaky Test Runner Stats
### 🟠 Some tests failed. - [kibana-flaky-test-suite-runner#8291](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/8291)
[❌] Security Solution Detection Engine - Cypress: 123/150 tests passed.

[see run history](https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds?branch=refs%2Fpull%2F220923%2Fhead)

denar50 commented on x-pack/test/security_solution_cypress/cypress/e2e/detection_response/detection_engine/exceptions/rule_details_flow/add_edit_exception_data_view.cy.ts:
```diff
@@ -48,7 +48,7 @@ import { waitForAlertsToPopulate } from '../../../../../tasks/create_new_rule';
 const DATAVIEW = 'auditbeat-exceptions-*';
 
 // Failing: See https://github.com/elastic/kibana/issues/207913
```
Will this comment be removed automatically? or should we do it manually in this PR?

rylnd replied:
I did miss this initially, but it was removed in a subsequent commit (you're just looking at a single commit, here).

denar50 reviewed (approved):
LGTM!

kibanamachine commented:
Starting backport for target branches: 8.17, 8.18, 8.19, 9.0

https://github.com/elastic/kibana/actions/runs/15218470223

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.17|Backport failed because of merge conflicts|
|❌|8.18|Backport failed because of merge conflicts|
|❌|8.19|Backport failed because of merge conflicts|
|❌|9.0|Backport failed because of merge conflicts|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 220923
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

rylnd commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|9.0|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/221437">](https://github.com/elastic/kibana/pull/221437)|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/221439">](https://github.com/elastic/kibana/pull/221439)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 10.0.0 -->

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
cc: @rylnd

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
cc: @rylnd

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
cc: @rylnd

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
cc: @rylnd
